magma/tuple.py

```diff
@@ -301,8 +301,13 @@ def wire(i, o, debug_info):
             o_elem = magma_value(o_elem)
             wire(o_elem, i_elem, debug_info)
 
-    def unwire(i, o):
-        for i_elem, o_elem in zip(i, o):
+    def unwire(self, o=None):
+        if o is None:
```

Comment:
any particular reason to do if-for here rather than for-if as in array? probably a little more readable to make them uniform